### PR TITLE
Processor updates

### LINKED ISSUE #11554: Investigate ways to avoid autocompleting when a user settles a message in handler
If a user leaves AutoComplete = true in the ServiceBusProcessorOptions, we will attempt to autocomplete assuming the handler doesn't trigger an exception. But what if the user settles the message themselves in some way - for instance Defering or DeadLettering the message? Currently, this would trigger a MessageLockLost exception that would bubble up to the user error handler. We should try to avoid this if possible.

### PR DESCRIPTION
**Pull request by JoshLove-msft**
- Don't autocomplete messages that have already been settled
- Remove lock renewal from processor event args
- Session Init/Close events

Fixes https://github.com/Azure/azure-sdk-for-net/issues/11554, https://github.com/Azure/azure-sdk-for-net/issues/11351, and https://github.com/Azure/azure-sdk-for-net/issues/11349

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs [hunk 1]
```diff
@@ -24,6 +25,12 @@ public class ProcessMessageEventArgs : EventArgs
         /// </summary>
         public CancellationToken CancellationToken { get; }
 
+        /// <summary>
+        /// Indicates whether the user has attempted to settle the message as part of their callback.
+        /// If they have done so, we will not autocomplete.
+        /// </summary>
+        internal bool UserSettled { get; set; }
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs:32]: Did you check if somehow we can access [Settled](https://github.com/Azure/azure-amqp/blob/master/src/Delivery.cs#L45) property after settling a message, then we don't need UserSettled property.

I've checked but couldn't find a way to access Settled property when we settle the messages :disappointed:

Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessMessageEventArgs.cs:32]: Can we rename UserSettled to something like isMessageSettled?

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs [hunk 2]
```diff
@@ -42,6 +42,12 @@ public class ProcessSessionMessageEventArgs : EventArgs
         /// </summary>
         public DateTimeOffset SessionLockedUntil => _sessionReceiver.SessionLockedUntil;
 
+        /// <summary>
+        /// Indicates whether the user has attempted to settle the message as part of their callback.
+        /// If they have done so, we will not autocomplete.
+        /// </summary>
+        internal bool UserSettled { get; set; }
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ProcessSessionMessageEventArgs.cs:48]: Can we rename UserSettled to something like isMessageSettled?

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs [hunk 3]
```diff
@@ -621,6 +731,10 @@ public virtual async Task StopProcessingAsync(CancellationToken cancellationToke
                         // 2. break out of the loop to allow requesting another session from the service
                         if (IsSessionProcessor && _sessionId == null)
                         {
+                            await OnSessionClosingAsync(
+                                (ServiceBusSessionReceiver)receiver,
+                                cancellationToken)
+                                .ConfigureAwait(false);
```
Comment (ShivangiReja) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs:737]: Lets say we create a receiver-1 and If we are done with receiving messages from 1 session, then create a new receiver and start receiving messages from second session and then we create receiver-3, receiver-4 and so on. How do we dispose all the receivers?

Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs:737]: The receiver gets disposed on [line 775](https://github.com/JoshLove-msft/azure-sdk-for-net/blob/processor-settle/sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs#L775)

Comment (ShivangiReja) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs:737]: Ah..finally block.

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**